app/src/org/commcare/activities/GlobalPrivilegeClaimingActivity.java

```diff
@@ -101,72 +105,45 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
         switch(requestCode) {
             case BARCODE_CAPTURE:
                 if (resultCode == RESULT_OK) {
-                    String[] fields = processScanResult(data.getStringExtra("SCAN_RESULT"));
-                    if (fields == null) {
-                        privilegeClaimFailed();
-                    } else {
-                        String flag = fields[0];
-                        String username = fields[1];
-                        String signature = fields[2];
-                        if (checkProperFormAndAuthenticity(flag, username, signature)) {
-                            GlobalPrivilegesManager.enablePrivilege(flag, username);
-                            refreshUI();
-                        } else {
-                            privilegeClaimFailed();
+                    String scanResult = data.getStringExtra("SCAN_RESULT");
+                    try {
+                        Pair<String, String[]> activatedPrivileges =
+                                new PrivilegesUtility(GlobalConstants.TRUSTED_SOURCE_PUBLIC_KEY).
+                                        processPrivilegePayloadForActivatedPrivileges(scanResult);
+
+                        for(String p : activatedPrivileges.second) {
+                            if (!GlobalPrivilegesManager.allGlobalPrivilegesList.contains(p)) {
```

Comment:
I know that you just copied this behavior from the code I had previously, but I'm now wondering why this is necessary and whether it might be harmful. Why would we want to prevent earlier versions of CommCare from being able to activate privileges that we hadn't yet conceived of at that point? This seems like it is severely limiting